src/mudlet-lua/lua/Other.lua

```diff
@@ -636,3 +636,29 @@ function shms(seconds, bool)
 		return hh, mm, ss
 	end
 end
+
+-- returns true if your Mudlet is older than the given version
+-- for example, it'll return true if you're on 2.1 and you do mudletOlderThan("3.1")
+-- it'll return false if you're on 4.0 and you do mudletOlderThan("4.0.0")
+-- credit to https://stackoverflow.com/a/16187766/72944 for the algorithm
+function mudletOlderThan(version)
+  local tonumber = tonumber
+	
+	-- strip trailing zeros from input version
+  local input = rex.gsub(version, [[(\.0+)+$]], ''):split('%.')
+	local mudlets = {getMudletVersion("table")}
+	
+	-- strip any build suffixes from Mudlets version so we only
+	-- have to deal with numbers in the comparison
+	if tonumber(mudlets[#mudlets]) == nil then mudlets[#mudlets] = nil end
+	
+	local minlength = math.min(#input, #mudlets)
+
+    local diff
+	for i = 1, minlength do
+	  diff = tonumber(input[i]) - mudlets[i]
```

Comment:
The `tonumber()` is there because string.split returns a list of strings. I don't think we should error on strings being passed as part of the version - it can happen - how about treating them as 0?

That'll also then remove the need to strip out -dev in the above lines.